### DISCUSSION TITLE
[Feature] | Ability to enable/disable the scheduler globally

### DISCUSSION THREAD

**Discussion by academy-codex** [jobrunr/jobrunr#705] [Ideas]
**Is your feature request related to a problem? Please describe.**
Sometimes when you are facing some critical issue in the jobs, you want to debug that LIVE but don't want to bring down your java services instances. For example, troubleshooting some live production issue. This becomes particularly interesting when you have a lot of recurring jobs scheduled.

**Describe the solution you'd like**
It would be very helpful if we could have a feature that allows us to toggle the scheduler on and off without affecting the application's overall functionality. This would be particularly useful when we need to temporarily suspend the scheduling of tasks, for example, during maintenance or troubleshooting activities.

**Describe alternatives you've considered**
Introduce a global flag across instances that would allow the user to enable/disable the scheduler.
A similar functionality is available across several open source cron based schedulers.

**Additional context**

**Comment by academy-codex** [comment]
@kfowler / @rdehuyss - Does this sound something like we can do in the JobRunr ? More than happy to pick it up :)

**Reply by rdehuyss** [reply]
Hi @academy-codex, I'm not completely following to be honest. 

For me, scheduling should work at all times. If I create a job, I want it to be created, otherwise data may get lot. What I think you mean (and what is already possible in the Pro version), is the pausing of all actual job processing (on one background job server or on all background job servers). This is indeed useful for maintenance, rolling updates, ... .

In JobRunr Pro, you can go to the servers tab inside the dashboard and pause job processing on all background job servers or only one background job server. And, there is also an API for this.

Is this what you are looking for? See also the screenshot below:
![image](https://user-images.githubusercontent.com/567842/220964506-3a79ce92-0a89-4cb6-9789-f3f85bfb5914.png)

**Reply by academy-codex** [reply]
Ah! Yup this is what i was mentioning. Any plans to roll out a lite version for this in the open source build ?

**Reply by rdehuyss** [reply]
I'm afraid not - I often hear that the oss version of JobRunr is already too good so I decided to add that feature to the Pro version only

And, wouldn't you want this project to be maintained for a long time? I know of a lot of open source projects that stop being maintained because giant corporations use their work without ever contributing back.

The same is also the case for JobRunr - there are a lot of big multinationals and fortune top 100 companies using it yet none of them take the Pro version. Even worse, some developers are trying to build pro features themselves (don't want to ask management for the Pro version or are not allowed to get a Pro subscription from management), make mistakes and bugs and then log an issue in this repository. This then often takes a lot of my (personal) time to figure out what is happening.

A lot of words to say that although I really like open-source, it really is tiring and consumes a lot of energy. I think we as a Java developer community are often too spoiled by the vast amount of great, free and open-source software that is out there often maintained by people in their free time. Yet, this also creates huge risks (remember log4shell ?).

> Any plans to roll out a lite version for this in the open source build ?

Depending on the Pro subscription, also JobRunr Pro is open-source (e.g. paying customers get access to the source code of the Pro version) 😉.

Sorry for my rant, I just hope you understand my point of view...

**Comment by rdehuyss** [comment]
It must be karma but this just appeared in my tweet: https://github.com/zloirock/core-js/blob/master/docs/2023-02-14-so-whats-next.md


![image](https://user-images.githubusercontent.com/567842/221017365-08497eba-5381-4efb-a79f-4307242b987f.png)
